src/set_theory/ordinal/arithmetic.lean

```diff
@@ -341,6 +341,24 @@ by rw [limit_rec_on, wf.fix_eq, dif_neg h.1, dif_neg (not_succ_of_is_limit h)];
 instance (o : ordinal) : order_top o.succ.out.α :=
 ⟨_, le_enum_succ⟩
 
+instance : unique (1 : ordinal).out.α :=
+{ default := enum (<) 0 (by { rwa type_lt, exact zero_lt_one }),
+  uniq := λ a, begin
+    change _ = enum _ 0 _,
+    rw ←enum_typein (<) a,
+    congr,
+    rw ←lt_one_iff_zero,
+    convert typein_lt_type (<) a,
+    rw type_lt
+  end }
+
+@[simp] theorem typein_lt_one_out (x : (1 : ordinal).out.α) : typein (<) x = 0 :=
+begin
+  rw unique.eq_default x,
+  unfold default,
+  apply typein_enum
+end
```

Comment:
does `by convert typein_enum _` with an appropriate number of underscores work here?

Reply:
Unfortunately no. It seems like the elaborator isn't able to find the proof that's built into the `enum` in the definition of `default`. However, an extra `simp` lemma makes this theorem almost immediate.